package/network/services/omcproxy/files/omcproxy.init

```diff
@@ -1,62 +1,88 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2010-2014 OpenWrt.org
+# Copyright (C) 2018 OpenWrt.org
 
 START=99
 USE_PROCD=1
 PROG=/usr/sbin/omcproxy
 
-# Uncomment to enable verbosity 
-#OPTIONS="-v" 
+# Uncomment to enable verbosity
+#OPTIONS="-v"
 PROXIES=""
 
-
 omcproxy_add_proxy() {
-	local uplink downlink scope proxy
-	config_get uplink $1 uplink
-	config_get downlink $1 downlink
+	local proxy scope uplink downlinks device up
+
 	config_get scope $1 scope
+	config_get uplink $1 uplink
+	config_get downlinks $1 downlink
+
+	[ -n "$uplink" ] || return
+
+	local status="$(ubus -S call "network.interface.$uplink" status)"
```

Comment:
network.sh (/lib/functions/network.sh) has wrapper functions which obsoletes the need to do ubus network calls in scripts

Reply:
Done.